### PR TITLE
[TRB-41904] upgrade dependency k8s.io/kubernetes

### PR DESCRIPTION
**Pull request by tian-ma**
# Intent

Fix scanner reported issues CVE-2023-2431, CVE-2023-2728, CVE-2023-2727
Caused by indirect dependencies from https://github.com/argoproj

# Background

Fix issues [TRB-41904](https://jsw.ibm.com/browse/TRB-41904), [TRB-42016](https://jsw.ibm.com/browse/TRB-42016), [TRB-42017](https://jsw.ibm.com/browse/TRB-42017)

Scanner report here: 
https://hyc-turbo-devops-team-jenkins.swg-devops.com/job/kubeturbo-whitesource-scanner/WhiteSource_20HTML_20Report/
The report recommends upgrading k8s.io/Kubernetes to v1.25.11
Trying higher upgrade (v1.27.3) breaks pkg dependencies from https://github.com/argoproj

# Testing

Check updated whitesource scanner report.
TwistLock scan [twistlock-scan-results-20230705-162317-565579000-UTC-06160807.results.csv](https://github.com/turbonomic/kubeturbo/files/11960159/twistlock-scan-results-20230705-162317-565579000-UTC-06160807.results.csv)

# Manual testing

- build image  using the changes

```bash
docker buildx build --platform=linux/amd64 --tag docker-na.artifactory.swg-devops.com/hyc-turbo-internal-team-docker-local/turbonomic/tamer/badkubeturbo:8.9.5-TM -f build/Dockerfile.multi-archs --build-arg VERSION=8.9.5-TM --push .
```
- deploy it to ROSA cluster using helm, test and check pod log for errors.
```yaml
image:
  #repository: icr.io/cpopen/turbonomic/kubeturbo
  repository:  docker-na.artifactory.swg-devops.com/hyc-turbo-internal-team-docker-local/turbonomic/tamer/badkubeturbo
  tag: 8.9.5-TM
  pullPolicy: Always
```

# Checklist

These are the items that must be done by the developer and by reviewers before the change is ready to merge. Please ~~strikeout~~ any items that are not applicable, but don't delete them

- [ ] Developer Checks
    - [X] Full build with unit tests and fmt and vet checks
    - [ ] ~~Unit tests added / updated~~
    - [X] No unlicensed images, no third-party code (such as from StackOverflow)
    - [ ] ~~Integration tests added / updated~~
    - [X] Manual testing done (and described)
    - [ ] Product sweep run and passed
    - [ ] ~~Developer wiki updated (and linked to this description)~~
- [ ] Reviewer Checks
    - [ ] Merge request description clear and understandable
    - [ ] Developer checklist items complete
    - [ ] Functional code review (how is the code written)
    - [ ] Architectural review (does the code try to do the right thing, in the right way)
    - [ ] Defensive coding (incoming data checked / sanitized, exceptions logged, clear error messages)
    - [ ] No unlicensed images, no third-party code (such as from StackOverflow)
    - [ ] Security review checklist complete.

# Audience

@ading1977 @irfanurrehman

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -44,7 +44,7 @@ require (
 	github.com/golang/protobuf v1.5.2 // indirect
 	github.com/google/go-querystring v1.1.0 // indirect
 	github.com/google/gofuzz v1.1.0 // indirect
-	github.com/google/uuid v1.1.2 // indirect
+	github.com/google/uuid v1.3.0 // indirect
```
Comment (pankajbatra1) [inline go.mod:47]: Why are google/uuid, x/net, and changes like that needed?

File: go.mod [hunk 2]
```diff
@@ -74,7 +74,7 @@ require (
 	gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7 // indirect
 	gopkg.in/yaml.v2 v2.4.0 // indirect
 	k8s.io/kube-openapi v0.0.0-20221012153701-172d655c2280 // indirect
-	k8s.io/kubernetes v1.25.4 // indirect
+	k8s.io/kubernetes v1.25.11 // indirect
```
Comment (pankajbatra1) [inline go.mod:77]: It would be good to start a Kubeturbo pod and do basic manual testing after these changes.

File: go.mod (hunk 1 above)
Comment (tian-ma) [reply go.mod:47]: those got updated when I updated the k8s.io/kubernetes

File: go.mod (hunk 2 above)
Comment (tian-ma) [reply go.mod:77]: Done. Description has been updated.
The pod was built, pushed to the private artifactory, deployed to ROSA and ran successfully.

**Review by ading1977 — APPROVED** [review]

**Pull request merged**